### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ip-ilya**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    cache_dict = {}
+
+    def inner(*args) -> Any:
```
Comment (Polyakiv-Andrey) [inline app/main.py:7]: Your function returns not any, but Callable

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+        nonlocal cache_dict
```
Comment (Polyakiv-Andrey) [inline app/main.py:8]: nonlocal in this task should be omitted

**Review by Polyakiv-Andrey — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (ip-ilya) [reply app/main.py:7]: but wht not `Any ` if  `return` cache_dict[args]` from inner can be int, list, str, etc. (depends on the type of *args)

**Review by Polyakiv-Andrey — APPROVED** [review]
Good job!

**Pull request closed without merging**